screwdriver/screwdriver.go

```diff
@@ -0,0 +1,185 @@
+package screwdriver
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// API has method to get job
+type API interface {
+	Job(jobName, filePath string) (Job, error)
+	JWT() string
+}
+
+// SDAPI has methods for control Screwdriver.cd APIs
+type SDAPI struct {
+	HTTPClient *http.Client
+	UserToken  string
+	APIURL     string
+	SDJWT      string
+}
+
+var _ API = (*SDAPI)(nil)
+
+// Step is step entity struct
+type Step struct {
+	Name    string `json:"name"`
+	Command string `json:"command"`
+}
+
+// Job is job entity struct
+type Job struct {
+	Steps       []Step            `json:"commands"`
+	Environment map[string]string `json:"environment"`
+	Image       string            `json:"image"`
+}
+
+// Jobs is Job entity map
+type Jobs map[string][]Job
+
+type validatorResponse struct {
+	Jobs   Jobs     `json:"jobs"`
+	Errors []string `json:"errors"`
+}
+
+type tokenResponse struct {
+	JWT string `json:"token"`
+}
+
+// New creates a SDAPI
+func New(apiURL, token string) (API, error) {
+	s := &SDAPI{
+		HTTPClient: http.DefaultClient,
+		APIURL:     apiURL,
+		UserToken:  token,
+	}
+
+	jwt, err := s.jwt()
+	if err != nil {
+		return s, err
+	}
+
+	s.SDJWT = jwt
+
+	return s, nil
+}
+
+func (sd *SDAPI) makeURL(path string) (*url.URL, error) {
+	version := "v4"
+	fullpath := fmt.Sprintf("%s/%s/%s", sd.APIURL, version, path)
+
+	return url.Parse(fullpath)
+}
+
+func (sd *SDAPI) request(method, path string, body io.Reader) (*http.Response, error) {
+	req, err := http.NewRequest(method, path, body)
+	if err != nil {
+		return nil, err
+	}
+
+	switch method {
+	case http.MethodGet:
+		{
+			req.Header.Add("Accept", "application/json")
+		}
+	case http.MethodPost, http.MethodPut, http.MethodDelete:
+		{
+			req.Header.Add("Content-Type", "application/json")
+			req.Header.Add("Authorization", "Bearer "+sd.SDJWT)
+		}
+	}
+
+	return sd.HTTPClient.Do(req)
+}
+
+func (sd *SDAPI) jwt() (string, error) {
+	path := "auth/token?api_token=" + sd.UserToken
+	fullpath, err := sd.makeURL(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to send request: %v", err)
```

Comment:
This case will not try to send request yet. And what kind of request is this cli sending? 
In my opinion, I think this msg is a little confusing.

Reply:
Fix the error message